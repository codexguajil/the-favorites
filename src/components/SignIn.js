import React, { Component } from 'react';
import {fetchData } from '../utils/fetch'

export class SignIn extends Component {
  constructor() {
    super()
    this.state = {
      email: '',
      password: ''
    }
  }

  handleChange = (e) => {
    const { name, value } = e.target;
    this.setState({ [name]: value })
  }

  handleSignIn = async (e) => {
    e.preventDefault();
    const url = 'http://localhost:3000/api/users'
    const userInput = this.state
    const userOptionObject = {
      method: "POST",
      body: JSON.stringify(userInput),
      headers: {
        "Content-Type": "application/json"
      }
    }
    const response = await fetchData(url, userOptionObject)
    console.log(response)
  }

  render() {
    return (
      <div>
        <form onSubmit={this.handleSignIn}>
          <input
            value={this.state.email}
            name='email'
            placeholder="Email"
            onChange={this.handleChange}
          />
          <input
            value={this.state.password}
            name='password'
            placeholder="Password"
            onChange={this.handleChange}
          />
          <button>Sign In</button>
        </form>
      </div>
    )
  }
}

export default SignIn
