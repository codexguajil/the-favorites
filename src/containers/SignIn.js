import React, { Component } from 'react';
import { fetchData } from '../utils/fetch';
import { cleanUsers } from '../utils/helpers';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { manageUser } from '../actions';

export class SignIn extends Component {
  constructor() {
    super()
    this.state = {
      name: '',
      email: '',
      password: '',
      status: ''
    }
  }

  handleChange = (e) => {
    const { name, value } = e.target;
    this.setState({ [name]: value })
  }

  handleAddUser = async (e) => {
    e.preventDefault();
    console.log('in submit', this.state)
    const url = 'http://localhost:3000/api/users/new#'
    const newUser = await fetchData(url, {
      method: 'POST',
      body: JSON.stringify(this.state),
      headers: {
        'Content-Type': 'application/json'
      }
    })
  }

  handleSignIn = async (e) => {
    e.preventDefault();
    const url = 'http://localhost:3000/api/users'
    const userInput = this.state
    const userOptionObject = {
      method: "POST",
      body: JSON.stringify(userInput),
      headers: {
        "Content-Type": "application/json"
      }
    }
    const response = await fetchData(url, userOptionObject)
    if(response.status === 'success') {
      this.props.manageUser(response.data)
    } 
    this.setState({ status: response.status })
  }

  render() {
    const { status } = this.state;
    return (
      <div>
        <Link to={'/'} className='back-btn'>Back To Home</Link>
        <form onSubmit={this.handleSignIn}>
        <input
            value={this.state.name}
            name='name'
            placeholder="your name"
            onChange={this.handleChange}
          />
          <input
            value={this.state.email}
            name='email'
            placeholder="Email"
            onChange={this.handleChange}
          />
          <input
            value={this.state.password}
            name='password'
            placeholder="Password"
            onChange={this.handleChange}
          />
          {status === 'success' &&
            <p>Welcome back!</p>
          }
          {status === 500 &&
            <p>Sorry, we couldn't find your account, please sign up.</p>
          }
          <button>Sign In</button>
          <button onClick={this.handleAddUser}>Sign Up</button>
        </form>
      </div>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  manageUser: (user) => dispatch(manageUser(user))
})

export default connect(null, mapDispatchToProps)(SignIn)
