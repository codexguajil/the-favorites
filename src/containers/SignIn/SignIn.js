import React, { Component } from 'react';
import { fetchData } from '../../utils/fetch';
import { cleanUsers, fetchAllFavorites } from '../../utils/helpers';
import { Link, Route, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import { logInUser, addAllFavorites, addMessage } from '../../actions';

export class SignIn extends Component {
  constructor() {
    super()
    this.state = {
      email: '',
      password: '',
      status: '',
    }
  }

  handleChange = (e) => {
    const { name, value } = e.target;
    this.setState({ [name]: value })
  }

  handleSignIn = async (e) => {
    const { addMessage } = this.props
    e.preventDefault();
    const url = 'http://localhost:3000/api/users'
    const userInput = this.state
    const userOptionObject = {
      method: "POST",
      body: JSON.stringify(userInput),
      headers: {
        "Content-Type": "application/json"
      }
    }
    const response = await fetchData(url, userOptionObject)
    if(response.status === 'success') {
      this.props.logInUser(response.data)
      const userFavorites = await fetchAllFavorites(response.data.id)
      this.props.addFavoritesToState(userFavorites)
      addMessage('Welcome Back.')
      setTimeout(() => {
        addMessage('')
      }, 3000)
    } 
    this.setState({ status: response.status })
  }
  
  render() {
    const { status } = this.state;
    return (
      <div className="login">
        <Link to={'/'} className='back-btn'>Back To Home</Link>
        <form onSubmit={this.handleSignIn} className="form">
          <h4>email:</h4>
          <input
            value={this.state.email}
            name='email'
            placeholder="Email"
            onChange={this.handleChange}
            className="input"
          />
          <h4>password</h4>
          <input
            value={this.state.password}
            name='password'
            placeholder="Password"
            onChange={this.handleChange}
            className="input"
          />
          {status === 'success' &&
            <Redirect to='/' />
          }
          {status === 500 &&
            <p>Sorry, we couldn't find your account, please sign up.</p>
          }
          <button>Sign In</button>
        </form>
      </div>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  logInUser: (user) => dispatch(logInUser(user)),
  addFavoritesToState: (favorite) => dispatch(addAllFavorites(favorite)),
  addMessage: (message) => dispatch(addMessage(message))
})

export default connect(null, mapDispatchToProps)(SignIn)
