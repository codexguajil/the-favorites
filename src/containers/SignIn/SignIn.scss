.login {
  background: hsla(360, 0%, 100%, 0.95);
  display: flex-box;
  font-family: 'arial';
  height: 500px;
  margin: auto;
  padding: 30px;
  width: 50%;
}

.form {
  background: hsla(0, 0%,10%, 1);
  display: flex;
  flex-direction: column;
  height: 80%;
  width: 40%;
  float: right;
  right: 15px;
}

.input {
  margin: auto;
}

h4 {
  color: hsla(360, 0%, 100%, 1);
}