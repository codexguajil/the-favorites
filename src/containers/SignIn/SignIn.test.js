import React from 'react';
import SignIn from './SignIn'
import { mapDispatchToProps } from './SignIn'
import { shallow } from 'enzyme'

describe('SignIn', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <SignIn />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('should update state when handleChange is invoked', () => {
  })

  describe('mapDispatchToProps', () => {
    it('should pass correct type when addFavoriteToStore is called', () => {
      const dispatch = jest.fn()

      mapDispatchToProps(dispatch).logInUser()
      expect(dispatch.mock.calls[0][0]).toEqual({ type: 'LOG_IN_USER' })
    });

    it('should pass correct type when deleteFavorite is called', () => {
      const dispatch = jest.fn()

      mapDispatchToProps(dispatch).addFavoritesToState()
      expect(dispatch.mock.calls[0][0]).toEqual({ type: 'ADD_ALL_FAVORITES' })
    })

    it('should pass correct type when addMessage is called', () => {
      const dispatch = jest.fn()

      mapDispatchToProps(dispatch).addMessage()
      expect(dispatch.mock.calls[0][0]).toEqual({ type: 'ADD_MESSAGE' })
    })
  })
})