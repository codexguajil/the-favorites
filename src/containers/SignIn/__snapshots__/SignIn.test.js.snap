// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should match snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
